nixos/modules/virtualisation/nixos-container.pl

```diff
@@ -68,6 +73,15 @@ sub showHelp {
 sub writeNixOSConfig {
     my ($nixosConfigFile) = @_;
 
+    my $localExtraConfig = "";
+
+    if ($extraConfig) {
+        $localExtraConfig = $extraConfig
+    } elsif ($configFile) {
```

Comment:
I see a potential problem here. If both `--config` and `--config-file` are provided, only the `--config` parameter will be respected and `--config-file` will be ignored silently, right? We should probably either merge `$localExtraConfig` or exit with an error.